### PR TITLE
Add ``WCSAxes.text_coord`` method

### PR DESCRIPTION
**Pull request by tbowers7**
### Description

This PR adds the ``WCSAxes.text_coord`` as a wrapper for ``Axes.text`` in the same way that the existing ``WCSAxes.plot_coord`` is a wrapper for ``Axes.plot``.

To facilitate adding text to figures with a WCS projection (_e.g._, for labeling objects in an image), this new method parses ``SkyCoord`` objects before calling ``Axes.text`` in the same way that ``WCSAxes.plot_coord`` does.

### CODE REVIEW

**Comment by pllim** [comment]
I think this is too late for v5.3 since feature freeze is tomorrow, so I put the milestone for v6. Also, could you please add a test?

**Comment by pllim** [comment]
@astrofrog or @Cadair will be able to advise you on how to deal with the CircleCI failures. Thanks for your patience!

**Comment by astrofrog** [comment]
See https://docs.astropy.org/en/latest/development/testguide.html#failing-tests for how to update the JSON hash files

File: astropy/visualization/wcsaxes/core.py [hunk 1]
```diff
@@ -358,6 +358,34 @@ def plot_coord(self, *args, **kwargs):
...
+
+        Parameters
+        ----------
+        coordinate : `~astropy.coordinates.SkyCoord` or `~astropy.coordinates.BaseCoordinateFrame`
+            The coordinate object to plot on the axes. This is converted to the
+            first two arguments to `matplotlib.axes.Axes.text`.
+
+        See Also
+        --------
+        matplotlib.axes.Axes.text :
```
Comment (pllim) [inline astropy/visualization/wcsaxes/core.py:381]: Is it more correct like this? I am not 100% sure.
```suggestion
        text : matplotlib.axes.Axes.text
```

Comment (larrybradley) [reply astropy/visualization/wcsaxes/core.py:381]: @pllim This is the "See Also" section, not "Parameters"/"Returns".

The trailing " :" isn't necessary here.

**Review by larrybradley — APPROVED** [review]
LGTM, thanks.

**Pull request merged**